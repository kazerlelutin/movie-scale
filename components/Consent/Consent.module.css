.box {
    position: fixed;
    top: 0;
    bottom:0;
    right: 0;
    left: 0;
    display: flex;
    justify-content: center;
    align-items: center;
    background: rgba(0, 0, 0, .5);
    z-index: 1000;

}

.container {
    border: 1px solid var(--color-mug);
    background: var(--color-round);
    box-shadow: 0 0 5px var(--color-mug), 0 0 15px var(--color-mug);
    padding: 15px;
    width: 100%;
    display: flex;
    max-width: 600px;
    margin: 20px;
    height: calc(100% - 40px);
    flex-direction: column;
    overflow-y: auto;
}

.desc {
    font-style: italic;
    margin: 10px 0;
}

.subtitle {
    color: var(--color-coffee);
    text-align: center;
    font-size: 1.3em;
}

.subtitle::before {
    content: "// ";
}

.buttons {
    margin: 15px;
    display: flex;
    align-items: center;
    gap: 10px;
    justify-content: space-evenly;
}

.buttons button {
    font-size: 1.2em;
}