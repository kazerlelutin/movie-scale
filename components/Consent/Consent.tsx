import { typeOfConsent } from "../../types/typeOfConsent.interface";
import classes from "./Consent.module.css";

export default function Consent({ setter }) {
  return (
    <div className={classes.box}>
      <section className={classes.container}>
        <div className="title">Nous utilisons des cookies</div>
        <article className={classes.article}>
          <div className={classes.subtitle}>Pour Votre session</div>
          <p>
            Si vous vous connectez à <span className="bold">Movie Scale</span>, 
            Vous acceptez les cookies nécessaires pour maintenir votre session active.
          </p>
          <p>
            Nous ne collectons que votre adresse email. 
          </p>
        </article>
        <article className={classes.article}>
          <div className={classes.subtitle}>Pour les statistiques</div>
          <p>Nous enregistrons les données de navigation.</p>
          <p>
          {"Pour suivre le trafic de l'application, nous utilisons "}
            <a href="https://fr.matomo.org/" target="_blank" rel="noreferrer">
              Matomo
            </a>
            .
          </p>
          <p>
            Ces données sont anonymes et nous hébergeons sur nos serveurs, notre
            propre instance Matomo.
          </p>
        </article>
        <article className={classes.article}>
          <div className={classes.subtitle}>Vous avez le choix</div>
          <div className={classes.buttons}>
            <div>Quittez le Movie Scale</div>
            <button onClick={()=>setter(typeOfConsent.NO_CONSENT)} >Refuser les cookies</button>
            <button onClick={()=>setter(typeOfConsent.CONSENT)}>Accepter les cookies</button>
          </div>
        </article>
      </section>
    </div>
  );
}
