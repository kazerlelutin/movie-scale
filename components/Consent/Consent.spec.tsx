import { render } from '@testing-library/react';
import Consent from './Consent';

describe('<Consent />', () => {

    it('Check snapshot', () => {
        const { asFragment } = render(
            <Consent setter={jest.fn()}/>
        );
        expect(asFragment()).toMatchSnapshot();
    });

});
