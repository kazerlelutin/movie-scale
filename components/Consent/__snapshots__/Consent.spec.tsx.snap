// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Consent /> Check snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <header
      class="header"
    >
      <a
        class="title"
        href="/"
      >
        f() 
        <span
          class="store"
        >
          Store
        </span>
      </a>
    </header>
    <main
      class="main"
    >
      <section
        class="container"
      >
        <div
          class="title"
        >
          Nous utilisons des cookies
        </div>
        <div
          class="desc"
        >
          Il est impossible d'utiliser le f() Store sans cookie, en voici les raisons:
        </div>
        <article
          class="article"
        >
          <div
            class="subtitle"
          >
            Pour la base de données
          </div>
          <p>
            Nous proposons un service sans base de données distantes.
          </p>
          <p>
            Toutes les informations que vous saisissez sont stockées sur votre navigateur.
          </p>
          <p>
            Aucune de ses informations sont enregistrées autrement.
          </p>
          <p>
            De cette façon, vous accédez aux outils sans avoir besoin de créer un compte.
          </p>
          <p>
            Aucune information personnelle n'est demandée, sauf si vous voulez utilisez une fonction qui requiert l'autorisation d'un tiers (connexion à Instagram par exemple)
          </p>
        </article>
        <article
          class="article"
        >
          <div
            class="subtitle"
          >
            Pour les statistiques
          </div>
          <p>
            Nous enregistrons les données de navigation.
          </p>
          <p>
            Pour suivre le trafic de l'application, nous utilisons 
            <a
              href="http://www.openwebanalytics.com/"
              target="_blank"
            >
              Open Web Analytics
            </a>
            .
          </p>
          <p>
            Ces données sont anonymes et nous hébergeons sur nos serveur, notre propre instance Open Web Analytics.
          </p>
        </article>
        <article
          class="article"
        >
          <div
            class="subtitle"
          >
            Vous avez le choix
          </div>
          <div
            class="buttons"
          >
            <div>
              Quittez le F() store
            </div>
            <button>
              Accepter les cookies
            </button>
          </div>
        </article>
      </section>
    </main>
  </div>
</DocumentFragment>
`;
